tests/CMakeLists.txt

```diff
@@ -47,6 +47,9 @@ add_executable(bpftrace_test
   ${CMAKE_SOURCE_DIR}/src/types.cpp
   ${CMAKE_SOURCE_DIR}/src/utils.cpp
   ${CMAKE_SOURCE_DIR}/src/disasm.cpp
+  ${CMAKE_SOURCE_DIR}/src/ast/semantic_analyser.cpp
+  ${CMAKE_SOURCE_DIR}/src/ast/codegen_llvm.cpp
+  ${CMAKE_SOURCE_DIR}/src/ast/irbuilderbpf.cpp
```

Comment:
We're already doing `target_link_libraries(bpftrace_test ast)` - doesn't that cover what we need without this change?

Reply:
I wanted the semantic and codegen tests to always run, regardless of whether the kernel supports `override_return` or not.

The way cgroupid solves that is by putting the `#ifdef` in the codegen, but then you loose the option to fail nicely (with line number and source code) and instead have to abort out, and you cannot test codegen.

The "hack" I did here is to always define `HAVE_OVERRIDE_RETURN` in the test suite, regardless whether the kernel supports it. And define `BPF_FUNC_override_return` when it is not supported by the system. As those flags are used in the semantic and codegen phases those have to be recompiled for the tests but not the actual bpftrace build.

This hack got already merged in with the merge of `signal` but I'll happily fix that if we can find a better way to do this.

Reply:
Makes sense. I can't think of a better way